File: ambeth/Ambeth.Security/ambeth/security/ISecurityManager.cs
using System;
using System.Collections.Generic;
using System.Reflection;

namespace De.Osthus.Ambeth.Security
{
    public interface ISecurityManager
    {
        void CheckMethodAccess(MethodInfo method, Object[] arguments, IAuthorization authorization);

        T FilterValue<T>(T value);
    }
}

File: jambeth/jambeth-security/src/main/java/de/osthus/ambeth/security/ISecurityManager.java
package de.osthus.ambeth.security;

import java.lang.reflect.Method;

import de.osthus.ambeth.security.SecurityContext.SecurityContextType;

public interface ISecurityManager
{
	void checkMethodAccess(Method method, Object[] arguments, SecurityContextType securityContextType, IUserHandle userHandle);

	<T> T filterValue(T value);
}
